qase-python-commons/src/qase/commons/client/api_v2_client.py

```diff
@@ -0,0 +1,148 @@
+                                      create_results_request_v2=CreateResultsRequestV2(results=results_to_send))
+        self.logger.log_debug(f"Results for run {run_id} sent successfully")
+
+    def _prepare_result(self, project_code: str, result: Result) -> ResultCreate:
+        attached = []
+        if result.attachments:
+            for attachment in result.attachments:
+                attached.extend(self._upload_attachment(project_code, attachment))
+
+        steps = []
+        for step in result.steps:
+            prepared = self._prepare_step(project_code, step)
+            steps.append(prepared)
+
+        for key, param in result.params.items():
+            # Hack to match old TestOps API
+            if param == "":
+                result.params[key] = "empty"
+
+        result_model_v2 = ResultCreate(
+            title=result.get_title(),
+            signature=result.signature,
+            testops_id=result.get_testops_id(),
+            execution=ResultExecution(start_time=result.execution.start_time, end_time=result.execution.end_time,
+                                      status=result.execution.status, duration=result.execution.duration,
+                                      stacktrace=result.execution.stacktrace, thread=result.execution.thread),
+            fields=result.fields,
+            attachments=[attach.hash for attach in attached],
+            steps=steps,
+            step_type=ResultStepsType.CLASSIC,
+            params=result.params,
+            muted=False,
+            message=result.message,
+        )
+
+        if result.get_suite_title():
+            data = []
+
+            for suite in result.get_suite_title().split("."):
+                data.append(RelationSuiteItem(title=suite))
+
+            result_model_v2.relations = ResultRelations(suite=RelationSuite(data=data))
+
+        self.logger.log_debug(f"Prepared result: {result_model_v2.to_json()}")
+
+        return result_model_v2
+
+    def _prepare_step(self, project_code: str, step: Step) -> Dict:
+        prepared_children = []
+
+        try:
+            prepared_step = {'execution': {}, 'data': {}, 'steps': []}
+            prepared_step['execution']['status'] = ResultStepStatus(step.execution.status)
+            prepared_step['execution']['duration'] = step.execution.duration
+
+            if step.step_type == StepType.TEXT:
+                prepared_step['data']['action'] = step.data.action
+                if step.data.expected_result:
+                    prepared_step['data']['expected_result'] = step.data.expected_result
+
+            if step.step_type == StepType.REQUEST:
+                prepared_step['data']['action'] = step.data.request_method + " " + step.data.request_url
+
+                if step.data.request_body:
+                    step.attachments.append(
+                        Attachment(file_name='request_body.txt', content=step.data.request_body, mime_type='text/plain',
```

Comment:
What if we have multiple requests? Will it work with several attachments having the same name?

Reply:
Yes, because the reporter will create separate step for each request. the second the code doesn't work and I will fix it.